samples/net/socket_echo/prj_qemu_x86.conf

```diff
@@ -0,0 +1,26 @@
+# General config
+CONFIG_NEWLIB_LIBC=y
+
+# Networking config
+CONFIG_NETWORKING=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_IPV6=n
+CONFIG_NET_TCP=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+
+# Network driver config
+CONFIG_NET_SLIP_TAP=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+
+# Without CONFIG_NET_BUF_LOG printf() doesn't work
```

Comment:
> I do not really understand how net_buf logging could affect printf(), could you elaborate the issue more.

I don't understand either, and yet it's like that - w/o CONFIG_NET_BUF_LOG=y, printf() output is not shown (network echo works). I did 10-15 mins of "genetic algorithm" debugging, semi-randomly trying other logging and output options and only CONFIG_NET_BUF_LOG looked like a culprit. Didn't debug further, as you may imagine, that can be quite a timesink ;-)